src/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixer.php

```diff
@@ -68,12 +73,18 @@ public function __construct()
         ];
 
         if (\PHP_VERSION_ID >= 80000) {
-            $this->hints = array_merge($this->hints, ['static' => true]);
-            $this->hints = array_merge($this->hints, ['mixed' => true]);
+            $this->hints['false'] = true;
```

Comment:
in tests, you say that `true/false/null` are added in 8.2, but here you compare with 8.0

what am I missing?

Reply:
@keradus In fact, @SpacePossum added it here in #6427 and this is because partial support for those types were [added in 8.0](https://wiki.php.net/rfc/union_types_v2) but only as return pseudo-type for unions. So technically it's not related to 8.2, but should be handled anyway 🙂

Reply:
We could make the fixer more accurate by only fixing the types when used as return type on version where applicable (>= 8.0 && < 8.2), but not sure worth the effort.

Reply:
Not sure we thing the same, so for PHP 8.0 the fixer should fix:

```php
function foo(STATIC $a): STATIC(){}
```

(so the return type, but _not_ the argument)

into 

```php
function foo(STATIC $a): static(){}
```

on PHP 8.2

```php
function foo(static $a): static(){}
```

(so _both_ return type and argument)

however, as PHP is case insensitive towards types it might not matter (a lot)